src/components/fileManager/FileManager.tsx

```diff
@@ -0,0 +1,121 @@
+import { TrashIcon } from '@heroicons/react/24/outline'
+import { useRouter } from 'next/router'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { useDropzone } from 'react-dropzone'
+import useGeneralTable from 'src/hooks/useGeneralTable'
+import useSWR from 'swr'
+
+import { Button } from '@/components/Buttons'
+import executeRequest from '@/helpers/executeRequest'
+import { uniFetcher } from '@/helpers/fetcher'
+
+export type FileItem = {
+  id: string
+  name: string
+  type: string
+  url: string
+}
+
+const FileManager = () => {
```

Comment:
Better approach will be for every usage configuration with props, not matching path with router. and even for that cas we have enums in /src/types/constants/paths.ts

Reply:
Could you explain why that would be the case?

Reply:
However, you're right. I'll need the `useRouter` hook anyways but this way we can check for proper configuration and throw error on mismatch.